samples/api/hello_triangle/hello_triangle.cpp

```diff
@@ -594,67 +646,55 @@ void HelloTriangle::init_render_pass()
-	subpass.pipelineBindPoint    = VK_PIPELINE_BIND_POINT_GRAPHICS;
-	subpass.colorAttachmentCount = 1;
-	subpass.pColorAttachments    = &color_ref;
+	VkSubpassDescription subpass{
+	    .pipelineBindPoint    = VK_PIPELINE_BIND_POINT_GRAPHICS,
+	    .colorAttachmentCount = 1,
+	    .pColorAttachments    = &color_ref,
+	};
 
 	// Create a dependency to external events.
 	// We need to wait for the WSI semaphore to signal.
 	// Only pipeline stages which depend on COLOR_ATTACHMENT_OUTPUT_BIT will
 	// actually wait for the semaphore, so we must also wait for that pipeline stage.
-	VkSubpassDependency dependency = {0};
-	dependency.srcSubpass          = VK_SUBPASS_EXTERNAL;
-	dependency.dstSubpass          = 0;
-	dependency.srcStageMask        = VK_PIPELINE_STAGE_COLOR_ATTACHMENT_OUTPUT_BIT;
-	dependency.dstStageMask        = VK_PIPELINE_STAGE_COLOR_ATTACHMENT_OUTPUT_BIT;
+	VkSubpassDependency dependency{
+	    .srcSubpass   = VK_SUBPASS_EXTERNAL,
+	    .dstSubpass   = 0,
+	    .srcStageMask = VK_PIPELINE_STAGE_COLOR_ATTACHMENT_OUTPUT_BIT,
+	    .dstStageMask = VK_PIPELINE_STAGE_COLOR_ATTACHMENT_OUTPUT_BIT};
 
 	// Since we changed the image layout, we need to make the memory visible to
 	// color attachment to modify.
 	dependency.srcAccessMask = 0;
 	dependency.dstAccessMask = VK_ACCESS_COLOR_ATTACHMENT_READ_BIT | VK_ACCESS_COLOR_ATTACHMENT_WRITE_BIT;
 
 	// Finally, create the renderpass.
-	VkRenderPassCreateInfo rp_info = {VK_STRUCTURE_TYPE_RENDER_PASS_CREATE_INFO};
-	rp_info.attachmentCount        = 1;
-	rp_info.pAttachments           = &attachment;
-	rp_info.subpassCount           = 1;
-	rp_info.pSubpasses             = &subpass;
-	rp_info.dependencyCount        = 1;
-	rp_info.pDependencies          = &dependency;
+	VkRenderPassCreateInfo rp_info{
+	    .sType           = VK_STRUCTURE_TYPE_RENDER_PASS_CREATE_INFO,
+	    .attachmentCount = 1,
+	    .pAttachments    = &attachment,
+	    .subpassCount    = 1,
+	    .pSubpasses      = &subpass,
+	    .dependencyCount = 1,
+	    .pDependencies   = &dependency};
 
 	VK_CHECK(vkCreateRenderPass(context.device, &rp_info, nullptr, &context.render_pass));
 }
 
 /**
- * @brief Helper function to load a shader module.
+ * @brief Helper function to load a shader module from an offline-compiled SPIR-V file
  * @param path The path for the shader (relative to the assets directory).
  * @returns A VkShaderModule handle. Aborts execution if shader creation fails.
  */
-VkShaderModule HelloTriangle::load_shader_module(const char *path)
+VkShaderModule HelloTriangle::load_shader_module(const std::string &path)
 {
-	vkb::GLSLCompiler glsl_compiler;
-
-	auto buffer = vkb::fs::read_shader_binary(path);
+	auto                  buffer = vkb::fs::read_shader_binary(path);
+	std::vector<uint32_t> spirv  = std::vector<uint32_t>(reinterpret_cast<uint32_t *>(buffer.data()),
+                                                        reinterpret_cast<uint32_t *>(buffer.data()) + buffer.size() / sizeof(uint32_t));
```

Comment:
Is the size of buffer always a multiple of 4 (`sizeof(uint32_t)`)?
Maybe add an assertion on it to express what's expected here.

Reply:
I did copy this from the framework's shader loading code, so I guess it's correct.